packages/logger/index.js

```diff
@@ -0,0 +1,123 @@
+const logger = require('../dd-trace/src/log')
+// path to require tracer logger
+
+const https = require('https')
+const coalesce = require('koalas')
+
+class V2LogWriter {
+  // Note: these attribute names match the corresponding entry in the JSON payload.
+  constructor ({ ddsource, hostname, service, site, apiKey, interval, timeout }) {
+    this.ddsource = ddsource
+    this.hostname = hostname
+    this.service = service
+    this.interval = interval
+    this.timeout = timeout
+    this.buffer = []
+    this.buffer_limit = 1000
+    this.apiKey = apiKey
+    this.endpoint = '/api/v2/logs'
+    this.site = coalesce(site, 'datadoghq.com')
+    this.intake = `http-intake.logs.${this.site}`
+    this.headers = {
+      'DD-API-KEY': this.apiKey,
+      'Content-Type': 'application/json'
+    }
+    this.timer = setInterval(() => {
+      this.flush()
+    }, this.interval)
+
+    logger.debug(`"started log writer to ${this.url}"`)
+  }
+
+  start () {
+    this.timer()
+  }
+
+  tagString (tags) {
+    let tagStr = ''
+    for (const key in tags) {
+      tagStr += key + ':' + tags[key] + ','
+    }
+    return tagStr
+  }
+
+  log (log, tags, span) {
+    const logTags = this.tagString(tags)
+    const toLog = {
+      'timestamp': Date.now(),
+      'message': log.message,
+      'hostname': coalesce(log.hostname, this.hostname),
+      'ddsource': coalesce(log.ddsource, this.ddsource),
+      'service': coalesce(log.service, this.service),
+      'status': log.level,
```

Comment:
Does this represent the "severity" of a log? Such as `info` or `warn`? If so it might be nice to introduce some helper methods:

```js
info(log, tags, span) {
  return this.log({status: 'info', ...log}, tags, span)
}
warn(log, tags, span) {
  return this.log({status: 'warn', ...log}, tags, span)
}
// etc.
```

This is a common convention across logger packages in the Node.js ecosystem and might help make the interface more familiar to developers. I think the standard levels are:

* TRACE
* DEBUG
* INFO
* WARN
* ERROR
* FATAL

Reply:
@tlhunter would the log itself contain the severity level? This would just be passed on from the log I would think, so the severity would be based on what the customer has in their logs 🤔 ?

Reply:
> would the log itself contain the severity level?

Yeah, it's just the `status` field in the final JSON that is sent.

In fact you're already making use of this pattern with the existing basic logger here:
https://github.com/DataDog/dd-trace-js/pull/3201/files#diff-fd61341f55194a3ae38d3f3cf77d04e3a9cdfe65122887e09eaefd985296aef7R29

Reply:
We can punt this feature for now.